tests/foreman/api/test_content_credentials.py

```diff
@@ -0,0 +1,233 @@
+"""Unit tests for the ``gpgkeys`` paths.
+
+:Requirement: ContentCredential
+
+:CaseAutomation: Automated
+
+:CaseLevel: Component
+
+:CaseComponent: ContentCredentials
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+import pytest
+from fauxfactory import gen_string
+from nailgun import entities
+from requests import HTTPError
+
+from robottelo.constants import VALID_GPG_KEY_BETA_FILE
+from robottelo.constants import VALID_GPG_KEY_FILE
+from robottelo.datafactory import invalid_values_list
+from robottelo.datafactory import parametrized
+from robottelo.datafactory import valid_data_list
+from robottelo.helpers import read_data_file
+
+
+key_content = read_data_file(VALID_GPG_KEY_FILE)
+
+
+@pytest.mark.tier1
+def test_positive_search_in_org(module_org):
+    """Search for a GPG key and specify just ``organization_id``.
+
+    :id: ff5e047c-404b-4379-8d28-3ad8cb39b6a9
+
+    :Steps:
+
+        1. Create an organization.
+        2. Create a GPG key belonging to the organization.
+        3. Search for GPG keys in the organization.
+
+    :expectedresults: Only one GPG key is in the search results: the
+        created GPG key.
+
+    :CaseImportance: Critical
+    """
+    org = entities.Organization().create()
+    gpg_key = entities.GPGKey(organization=org).create()
+    gpg_keys = gpg_key.search({'organization'})
+    assert len(gpg_keys) == 1
+    assert gpg_key.id == gpg_keys[0].id
+
+
+@pytest.mark.parametrize('name', **parametrized(valid_data_list()))
+@pytest.mark.tier1
+def test_positive_create_with_name(module_org, name):
+    """Create a GPG key with valid name.
+
+    :id: 741d969b-28ef-481f-bcf7-ed4cd920b030
+
+    :expectedresults: A GPG key is created with the given name.
+
+    :CaseImportance: Critical
+    """
+    gpg_key = entities.GPGKey(organization=module_org, name=name).create()
+    assert name == gpg_key.name
+
+
+@pytest.mark.tier1
+def test_positive_create_with_content(module_org):
+    """Create a GPG key with valid name and valid gpg key text.
+
+    :id: cfa6690e-fed7-49cf-94f9-fd2deed941c0
+
+    :expectedresults: A GPG key is created with the expected content.
+
+    :CaseImportance: Critical
+    """
+    gpg_key = entities.GPGKey(organization=module_org, content=key_content).create()
+    assert key_content == gpg_key.content
+
+
+@pytest.mark.parametrize('name', **parametrized(invalid_values_list()))
+@pytest.mark.tier1
+def test_negative_create_name(module_org, name):
+    """Attempt to create GPG key with invalid names only.
+
+    :id: 904a3ed0-7d50-495e-a700-b4f1ae913599
+
+    :expectedresults: A GPG key is not created and error is raised.
+
+    :CaseImportance: Critical
+    """
+    with pytest.raises(HTTPError):
+        entities.GPGKey(name=name).create()
```

Comment:
I think you missed the module_org fixture.

Reply:
but test passes:

robottelo]$ pytest -s tests/foreman/api/test_content_credentials.py -k test_negative_create_name
============================= test session starts =====================================
platform linux -- Python 3.8.6, pytest-5.4.3, py-1.9.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/sjw/sat6/robottelo_1/robottelo, inifile: pytest.ini
plugins: ibutsu-1.1.1, mock-3.3.1, forked-1.3.0, services-2.2.1, cov-2.10.1, xdist-1.34.0
collecting ... 2020-11-12 08:42:07 - conftest - DEBUG - Collected 42 test cases
collected 42 items / 32 deselected / 10 selected
==================== 10 passed, 32 deselected, 2 warnings in 52.01s ======================

Reply:
"Module org" is the fixtures that we use for organization creation on the module level, if we are not using the module level organization for this test case then for every check the new organization will be created and execution times get increased.

Using moudule module_org.
```
$ pytest tests/foreman/api/test_content_credentials.py::test_negative_create_name
=============================================================================================== test session starts ===============================================================================================                                                                                                                                                                  

tests/foreman/api/test_content_credentials.py ..........                                                                                                                                                    [100%]
======================================================================================== 10 passed, 14 warnings in 26.35s =========================================================================================
```
Execution without module_org fixture.
```
$ pytest tests/foreman/api/test_content_credentials.py::test_negative_create_name
=============================================================================================== test session starts ===============================================================================================                                                                                                                                                                                           

tests/foreman/api/test_content_credentials.py ..........                                                                                                                                                    [100%]

================================================================================================ warnings summary =================================================================================================

=================================================================================== 10 passed, 14 warnings in 157.21s (0:02:37) =================================================================================== 
```